Add HP Smart Storage Array python plugin (#7181)
netdata/netdata: collectors/python.d.plugin/hpssa/README.md

# hpssa

> Module collects controller, logical and physical devices health metrics from HP Smart Storage Arrays.

Reviewer: ```suggestion
This module collects controller, logical, and physical device health metrics from HP Smart Storage Arrays.
```

## Requirements:

- `ssacli` program
- `sudo` program
- `netdata` user needs to be able to sudo the `ssacli` program without password

> To grab stats it executes: `sudo -n ssacli ctrl all show config detail`

Reviewer: ```suggestion
To collect metrics, the module executes: `sudo -n ssacli ctrl all show config detail`
```

> It produces:

Reviewer: ```suggestion
This module produces:
```

1. Controller state and temperature
2. Cache module state and temperature
3. Logical drive state

> 4. Physical drive state and temperature

Reviewer: I know it's a little strange, but our linter wants two extra spaces between the period and the start of the text.

So:

```
1.   Controller state and temperature
2.   Cache module state and temperature
3.   Logical drive state
4.   Physical drive state and temperature
```

## Prerequisite

> This module uses `ssacli` which can only be executed by root.  It uses

Reviewer: ```suggestion
This module uses `ssacli`, which can only be executed by root. It uses
```

`sudo` and assumes that it is configured such that the `netdata` user can
execute `ssacli` as root without password.

Add to `sudoers`:

```
netdata ALL=(root)       NOPASSWD: /path/to/ssacli
```

## Configuration

**hpssa** is disabled by default. Should be explicitly enabled in `python.d.conf`.

```yaml
hpssa: yes
```

> If `ssacli` cannot be found in the `PATH`, configure it in `hpssa.conf`

Reviewer: ```suggestion
If `ssacli` cannot be found in the `PATH`, configure it in `hpssa.conf`.
```
